### ISSUE TITLE
redefines-builtin-id: `comparable` is not reported

### ISSUE THREAD

**Issue by dylanpulver** [mgechev/revive#1785]
# Bug report

## Describe the bug

`redefines-builtin-id` does not report redefinitions of `comparable`.

`comparable` is a predeclared identifier in the Go spec, listed on the same
line of "Predeclared identifiers → Types" as `any`, which the rule *does*
report. It is the only member of `go/types.Universe` the rule's three tables
miss:

```
universe=44 revive=43
MISSING from revive: [comparable]
EXTRA in revive:     []
```

(union of `builtInConstAndVars` + `builtFunctions` + `builtFunctionsAfterGo121`
+ `builtInTypes` in `rule/redefines_builtin_id.go`, diffed against
`go/types.Universe.Names()` on Go 1.26.)

Shadowing `comparable` makes the `comparable` constraint unreachable in that
scope — exactly the "bugs very difficult to detect" the rule documents.

## Steps to reproduce

`main.go`:

```go
package main

type comparable int

type any int

func withTypeParam[comparable any]() {}
```

```shell
revive --config revive.toml ./main.go
```

```toml
[rule.redefines-builtin-id]
```

## Actual behavior

```
main.go:5:1: redefinition of the built-in type any
```

Nothing is reported for either `comparable` declaration.

## Expected behavior

`comparable` reported the same way `any` is, in all declaration forms the rule
already covers (type/func/var/const/type-param).

## Additional context

The `builtInTypes` table has been touched twice: #659 added `any`, and #675
removed `Type`/`Type1`/`IntegerType`/`FloatType`/`ComplexType` as
documentation-only pseudo-types. `comparable` is neither — it was simply never
added when `any` was. Same shape as #1021/#1024 for `clear`/`max`/`min`.

Happy to send a PR (one entry in `builtInTypes` + fixture cases mirroring the
existing `any` block).

---

Drafted with AI assistance (Claude Code, claude-opus-5); the differential, the repro and the fence history were verified by hand against the Go spec and `go/types.Universe`.
